models/JSONPrettyPrint.cfc

```diff
@@ -4,147 +4,59 @@
 * www.coldbox.org | www.ortussolutions.com
 ********************************************************************************
 * @author Brad Wood, Luis Majano
-* 
+*
 */
 component accessors="true" singleton alias='JSONPrettyPrint' {
 
-	function init() {
-		variables.os = createObject( "java", "java.lang.System" ).getProperty( "os.name" ).toLowerCase();
-		return this;
-	}
-		
-	// OS detector
-	private boolean function isWindows(){ return variables.os.contains( "win" ); }
+    property CFMLPrinter;
+    property JSONPrinter;
+    property DefaultPrinter;
 
-	/**
-	 * Pretty JSON
-	 * @json A string containing JSON, or a complex value that can be serialized to JSON
-	 * @indent String to use for indenting lines.  Defaults to four spaces.
-	 * @lineEnding String to use for line endings.  Defaults to CRLF on Windows and LF on *nix
-	 * @spaceAfterColon Add space after each colon like "value": true instead of"value":true 
- 	 **/
-	public function formatJson( any json, string indent='    ', lineEnding, boolean spaceAfterColon=false ) {
-		
-		// Default line ending based on OS
-		if( isNull( arguments.lineEnding ) ) {
-			if( isWindows() ) {
-				arguments.lineEnding = chr( 13 ) & chr( 10 );
-			} else {
-				arguments.lineEnding = chr( 10 );				
-			}
-		}
-		
-		// Overload this method to accept a struct or array
-		if( !isSimpleValue( arguments.json ) ) {
-			arguments.json = serializeJSON( arguments.json );
-		}
-		
-		arguments.json = arguments.json.replace( '	', '', 'all' );
-		arguments.json = arguments.json.replace( chr( 10 ), '', 'all' );
-		arguments.json = arguments.json.replace( chr( 13 ), '', 'all' );
-		
-		var retval = createObject( 'java', 'java.lang.StringBuilder' ).init( '' );
-		var str = json;
-		var strLen = str.len();
-	    var pos = 0;
-	    var strLen = str.length();
-		var indentStr = arguments.indent;
-	    var newLine = arguments.lineEnding;
-		var char = '';
-		var nextChar = '';
-		var inQuote = false;
-		var isEscaped = false;
-		var itemsInCollection = [];
+    function init( string defaultPrinter = '' ) {
+        if ( !len( defaultPrinter ) ) {
+            if ( server.coldfusion.productname == 'Lucee' && listFirst( server.lucee.version, '.' ) >= 5 ) {
+                defaultPrinter = 'CFMLPrinter';
+            } else {
+                defaultPrinter = 'JSONPrinter';
+            }
+        }
+        setJSONPrinter( new JSONPrinter() );
+        setCFMLPrinter( new CFMLPrinter() );
```

Comment:
I deliberately (selfishly) avoided Wirebox here so that I could use this library in a non Wirebox environment. If you want Wirebox involved I can use it here, but that won't allow reuse without Wirebox. (This is your library, and is is a ColdBox module, so I understand if that is what you want.)

Reply:
Hmm, I see.  Once you start manually creating CFCs, it's a dark road that leads to destruction, especially when the rest of your app uses WireBox :)  I gave up long ago even trying to make modules cross the divide, it's just too limiting and the features of power of WireBox are too great to even both with the old ways.  That's for me, anyway.  This isn't going to kill anyone to leave it this way but you're basically screwed as soon as either of those CFCs get more complicated to the point where _they_ need things injected into them or persistance like singleton.  An in-between solution is to put in the properties to inject which will get picked up and processed automatically when WireBox creates this CFC, and then make non-WireBox code call the setters manually to pass the instances.  I think that's what Eric did on qb.